tds_test.go

```diff
@@ -665,6 +665,28 @@ func TestSecureConnection(t *testing.T) {
 	}
 }
 
+func TestTDS8ConnFailure(t *testing.T) {
+	checkConnStr(t)
+	tl := testLogger{t: t}
+	defer tl.StopLogging()
+	SetLogger(&tl)
+	config := testConnParams(t)
+	dsn := config.URL()
+	if !strings.HasSuffix(strings.Split(dsn.Host, ":")[0], ".database.windows.net") {
+		t.Skip()
+	}
+	dsnParams := dsn.Query()
+	dsnParams.Set(msdsn.TrustServerCertificate, "true")
```

Comment:
>t(msdsn.TrustServerCertificate, "true") [](http://example.com/codeflow?start=13&length=39)

this is counterintuitive. if I say  `TrustServerCertificate=true` why is it ignored? Does SqlClient do the same thing?

Reply:
As tested on ADS, with encrypt=strict, 'TrustServerCertificate=true' does not allow the connection.